### PR TITLE
Rename `brew man` to `brew generate-man-completions`

### PR DESCRIPTION
**Pull request by nandahkrishna**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

This PR renames `brew man` to ~`brew generate-manpages`~ `brew generate-man-completions`, to hopefully avoid confusion between the developer command to (re-)generate manpages and `man brew`.

### CODE REVIEW

**Review by MikeMcQuaid — APPROVED** [review]

**Review by Rylan12 — APPROVED** [review]
Looks good to me.

The only thing I'd ask is whether there's a better name than `generate-manpages` because the command also generates shell completions. I couldn't figure out what the command _should_ be called, though, so 👍 as is unless anyone can think of the right name.

**Comment by SMillerDev** [comment]
> The only thing I'd ask is whether there's a better name than generate-manpages because the command also generates shell completions.

`brew brew`? 😄

**Comment by gromgit** [comment]
`brew make-man`? :grinning: 

But seriously, if it makes more than man pages, perhaps `update-support`?

**Comment by carlocab** [comment]
> `brew brew`? 😄

The "request changes" button has never looked so tempting.

**Comment by MikeMcQuaid** [comment]
> The only thing I'd ask is whether there's a better name than `generate-manpages` because the command also generates shell completions. I couldn't figure out what the command _should_ be called, though, so 👍 as is unless anyone can think of the right name.

`brew generate-manpages-completions`?

**Comment by nandahkrishna** [comment]
I think that's a bit too long, even though we have completions to autocomplete the command. I'm inclined to stick with `generate-manpages` or `update-manpages`, but how about `update-man-completions`?

**Comment by MikeMcQuaid** [comment]
> I think that's a bit too long, even though we have completions to autocomplete the command.

These two things contradict 😉 



> but how about `(update|generate)-man-completions`?

This is shorter so I'm fine with it. I do agree that if we're gonna update the name now: may as well do what it says more accurately.

**Comment by Rylan12** [comment]
Maybe go with `generate-man-completions` because that way I can

```console
$ brew ge<TAB>
```

There are already 8 commands that start with `up`...

File: Library/Homebrew/dev-cmd/generate-man-completions.rb [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+
+  SOURCE_PATH = (HOMEBREW_LIBRARY_PATH/"manpages").freeze
+  TARGET_MAN_PATH = (HOMEBREW_REPOSITORY/"manpages").freeze
+  TARGET_DOC_PATH = (HOMEBREW_REPOSITORY/"docs").freeze
+
+  sig { returns(CLI::Parser) }
+  def generate_man_completions_args
+    Homebrew::CLI::Parser.new do
+      description <<~EOS
+        Generate Homebrew's manpages.
```
Comment (Rylan12) [inline Library/Homebrew/dev-cmd/generate-man-completions.rb:23]:
```suggestion
        Generate Homebrew's manpages and shell completions.
```

**Review by Rylan12 — DISMISSED** [review]

**Review by Rylan12 — APPROVED** [review]

**Pull request merged**